packages/grafana-ui/src/components/TimePicker/TimeRangePicker/TimePickerContent.tsx

```diff
@@ -1,181 +1,191 @@
 
-const getStyles = stylesFactory((theme: GrafanaTheme, isReversed) => {
-  const containerBorder = theme.isDark ? theme.palette.dark9 : theme.palette.gray5;
-
-  return {
-    container: css`
-      background: ${theme.colors.bodyBg};
-      box-shadow: 0px 0px 20px ${theme.colors.dropdownShadow};
-      position: absolute;
-      z-index: ${theme.zIndex.modal};
-      width: 546px;
-      top: 116%;
-      border-radius: 2px;
-      border: 1px solid ${containerBorder};
-      right: ${isReversed ? 'unset' : 0};
-
-      @media only screen and (max-width: ${theme.breakpoints.lg}) {
-        width: 262px;
-      }
-    `,
-    body: css`
-      display: flex;
-      height: 381px;
-    `,
-    leftSide: css`
-      display: flex;
-      flex-direction: column;
-      border-right: ${isReversed ? 'none' : `1px solid ${theme.colors.border1}`};
-      width: 60%;
-      overflow: hidden;
-      order: ${isReversed ? 1 : 0};
-
-      @media only screen and (max-width: ${theme.breakpoints.lg}) {
-        display: none;
-      }
-    `,
-    rightSide: css`
-      width: 40% !important;
-      border-right: ${isReversed ? `1px solid ${theme.colors.border1}` : 'none'};
-
-      @media only screen and (max-width: ${theme.breakpoints.lg}) {
-        width: 100% !important;
-      }
-    `,
-    spacing: css`
-      margin-top: 16px;
-    `,
-  };
-});
-
-const getNarrowScreenStyles = stylesFactory((theme: GrafanaTheme) => {
-  const formBackground = theme.isDark ? theme.palette.gray15 : theme.palette.gray98;
-
-  return {
-    header: css`
-      display: flex;
-      flex-direction: row;
-      justify-content: space-between;
-      align-items: center;
-      border-bottom: 1px solid ${theme.colors.border1};
-      padding: 7px 9px 7px 9px;
-    `,
-    body: css`
-      border-bottom: 1px solid ${theme.colors.border1};
-      background: ${formBackground};
-      box-shadow: inset 0px 2px 2px ${theme.colors.dropdownShadow};
-    `,
-    form: css`
-      padding: 7px 9px 7px 9px;
-    `,
-  };
-});
+interface Fits {
+  left: boolean;
+  right: boolean;
+  top: boolean;
+  bottom: boolean;
+}
 
-const getFullScreenStyles = stylesFactory((theme: GrafanaTheme) => {
-  return {
-    container: css`
-      padding-top: 9px;
-      padding-left: 11px;
-      padding-right: 20%;
-    `,
-    title: css`
-      margin-bottom: 11px;
-    `,
-    recent: css`
-      flex-grow: 1;
-      display: flex;
-      flex-direction: column;
-      justify-content: flex-end;
-    `,
-  };
-});
+const checkIfFits = (windowWidth: number, windowHeight: number, bounding?: DOMRect): Fits => {
+  if (!bounding) {
+    return {
+      left: true,
+      right: true,
+      bottom: true,
+      top: true,
+    };
+  }
 
-const getEmptyListStyles = stylesFactory((theme: GrafanaTheme) => {
-  const formBackground = theme.isDark ? theme.palette.gray15 : theme.palette.gray98;
+  const { bodyTallHeight, containerWideWidth, calendarWidth } = stylesVariables;
```

Comment:
This makes this function tightly coupled with `stylesVariables` global variable, should it be passed as an argument instead?